app/models/plate_purpose/input.rb

```diff
@@ -10,10 +12,46 @@
 #       flexibility, such as by allowing library-manifest plates to pass straight in to the
 #       pipeline.
 class PlatePurpose::Input < PlatePurpose
-  include PlatePurpose::Stock
+  self.state_changer = StateChanger::InputPlate
 
+  UNREADY_STATE  = 'pending'
+  READY_STATE    = 'passed'
   WELL_STATE_PRIORITY = %w[pending started passed failed cancelled].freeze
 
```

Comment:
All the additions here are inlined from PlatePurpose::Stock as this was the only place it was used.